tests/modules/images/image-loading-optimization/optimization-tests.php

```diff
@@ -286,6 +358,124 @@ public function data_provider_test_ilo_optimize_template_output_buffer(): array
+						for ( $i = 0; $i < $sample_size; $i++ ) {
+							ilo_store_url_metric(
+								home_url( '/' ),
+								$slug,
+								$this->get_validated_url_metric(
+									$viewport_width,
+									array(
+										array(
+											'xpath' => '/*[0][self::HTML]/*[1][self::BODY]/*[0][self::DIV]',
+											'isLCP' => true,
+										),
+									)
+								)
+							);
+						}
+					}
+				},
+				'buffer'   => '
+					<html lang="en">
+						<head>
+							<meta charset="utf-8">
+							<title>...</title>
+						</head>
+						<body>
+							<div style="background-image:url(https://example.com/foo-bg.jpg); width:100%; height: 200px;">This is so background!</div>
+						</body>
+					</html>
+				',
+				'expected' => '
+					<html lang="en">
+						<head>
+							<meta charset="utf-8">
+							<title>...</title>
+							<link as="image" data-ilo-added-tag="" fetchpriority="high" href="https://example.com/foo-bg.jpg" rel="preload" media="screen">
+						</head>
+						<body>
+							<div style="background-image:url(https://example.com/foo-bg.jpg); width:100%; height: 200px;">This is so background!</div>
+						</body>
+					</html>
+				',
+			),
+
+			'responsive-background-images'                => array(
+				'set_up'   => function () {
+					$mobile_breakpoint  = 480;
+					$tablet_breakpoint  = 600;
+					$desktop_breakpoint = 782;
+					add_filter(
+						'ilo_breakpoint_max_widths',
+						static function () use ( $mobile_breakpoint, $tablet_breakpoint ): array {
+							return array( $mobile_breakpoint, $tablet_breakpoint );
+						}
+					);
+					$sample_size = ilo_get_url_metrics_breakpoint_sample_size();
+
+					$slug = ilo_get_url_metrics_slug( ilo_get_normalized_query_vars() );
+					$div_index_to_viewport_width_mapping = array(
+						0 => $desktop_breakpoint,
+						1 => $tablet_breakpoint,
+						2 => $mobile_breakpoint,
+					);
+
+					foreach ( $div_index_to_viewport_width_mapping as $div_index => $viewport_width ) {
+						for ( $i = 0; $i < $sample_size; $i++ ) {
+							ilo_store_url_metric(
+								home_url( '/' ),
+								$slug,
+								$this->get_validated_url_metric(
+									$viewport_width,
+									array(
+										array(
+											'xpath' => "/*[0][self::HTML]/*[1][self::BODY]/*[{$div_index}][self::DIV]",
+											'isLCP' => true,
+										),
+									)
+								)
+							);
+						}
+					}
+				},
+				'buffer'   => '
+					<html lang="en">
+						<head>
+							<meta charset="utf-8">
+							<title>...</title>
+							<style>/* responsive styles to show only one div.header at a time... */</style>
+						</head>
+						<body>
+							<div class="header desktop" style="background: red no-repeat center/80% url(\'https://example.com/desktop-bg.jpg\'); width:100%; height: 200px;">This is the desktop background!</div>
+							<div class="header tablet" style=\'background-image:url( "https://example.com/tablet-bg.jpg" ); width:100%; height: 200px;\'>This is the tablet background!</div>
```

Comment:
the quoting on this line is reversed from all of the others which makes it stand out. is that required for the test to pass?

Reply:
It's just testing that single-quoted and double-quoted strings both work in the CSS `url()`.